app/services/TwoStepVerificationThrottle.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import org.joda.time.DateTime
+import play.api.Play.{current, _}
+
+trait TwoStepVerificationThrottle {
+  def hourlyLimit: HourlyLimit
+
+  def registrationMandatory(discriminator: String) = {
+    val userValue = Math.abs(discriminator.hashCode % 100)
+    val threshold = hourlyLimit.getCurrentLimit
+    userValue <= threshold
+  }
+}
+
+object TwoStepVerificationThrottle extends TwoStepVerificationThrottle {
+  override lazy val hourlyLimit = HourlyLimit
+}
+
+
+trait HourlyLimit {
```

Comment:
naming: HourlyLimit suggests that there is a limit applied per hour - which is not the case. I suggest calling it TimeBasedLimit or similar